connectors/src/connectors/notion/lib/parents.ts

```diff
@@ -116,39 +125,33 @@ export async function updateAllParentsFields(
  * updated
  */
 async function getPagesToUpdate(
-  pageOrDbs: (NotionPage | NotionDatabase)[],
+  createdOrMovedNotionPageIds: string[],
+  createdOrMovedNotionDatabaseIds: string[],
   dataSourceConfig: DataSourceConfig
-): Promise<NotionPage[]> {
-  const pagesToUpdate: NotionPage[] = [];
+): Promise<Set<string>> {
+  const pageIdsToUpdate: Set<string> = new Set([
+    ...createdOrMovedNotionPageIds,
+  ]);
 
-  let i = 0;
-  while (i < pageOrDbs.length) {
-    // Visit next document and if it's a page add it to update list
-    const pageOrDb = pageOrDbs[i++] as NotionPage | NotionDatabase;
-    const pageOrDbId = notionPageOrDbId(pageOrDb);
-    if ((pageOrDb as NotionPage).notionPageId) {
-      pagesToUpdate.push(pageOrDb as NotionPage);
-    }
+  const toUpdate = [
+    ...createdOrMovedNotionPageIds,
+    ...createdOrMovedNotionDatabaseIds,
+  ];
 
-    // Get children of the document
+  for (const pageOrDbId of toUpdate) {
     const pageChildren = await getPageChildrenOf(dataSourceConfig, pageOrDbId);
     const databaseChildren = await getDatabaseChildrenOf(
       dataSourceConfig,
       pageOrDbId
     );
 
-    // If they haven't yet been visited, add them to documents visited
-    // and to the list of documents whose children should be fetched
     for (const child of [...pageChildren, ...databaseChildren]) {
-      if (
-        !pageOrDbs.some((d) => notionPageOrDbId(d) === notionPageOrDbId(child))
-      ) {
-        pageOrDbs.push(child);
-      }
+      const childId = notionPageOrDbId(child);
+      pageIdsToUpdate.add(childId);
```

Comment:
it seems that databases are added also, while only pages should be added to this